File: java/src/main/java/com/sasteval/safe/FrameworkEscaped.java
package com.sasteval.safe;

import org.owasp.html.Sanitizers;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * TRUE NEGATIVE for CWE-79: XSS
 * SAFE: User input is sanitized using OWASP HTML Sanitizer with FORMATTING and BLOCKS policies.
 */
public class FrameworkEscaped extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String input = request.getParameter("input");
        response.setContentType("text/html");

        // SAFE: OWASP sanitizer strips dangerous HTML
        String sanitized = Sanitizers.FORMATTING.and(Sanitizers.BLOCKS).sanitize(input);
        response.getWriter().println("<div>" + sanitized + "</div>");
    }
}

File: dotnet/Safe/FrameworkEscaped.cs
using Ganss.Xss;

namespace SastEval.Safe;

/// <summary>
/// True Negative for CWE-79: User input is sanitized via HtmlSanitizer before rendering.
/// SAST tools should NOT flag this as XSS.
/// </summary>
public static class FrameworkEscaped
{
    private static readonly HtmlSanitizer _sanitizer = new();

    public static IResult Handle(HttpContext context)
    {
        var userInput = context.Request.Query["input"].ToString();

        if (string.IsNullOrEmpty(userInput))
        {
            return Results.BadRequest("Missing 'input' query parameter");
        }

        // SAFE: HtmlSanitizer strips dangerous tags and attributes before rendering
        var sanitized = _sanitizer.Sanitize(userInput);

        return Results.Content(
            $"<html><body><h1>User Profile</h1><div>{sanitized}</div></body></html>",
            "text/html"
        );
    }
}
